src/NuGet.Core/NuGet.Packaging/PackageExtractor.cs

```diff
@@ -995,7 +995,19 @@ public static class PackageExtractor
                        token,
                        parentId);
 
-                if (!verifyResult.Valid)
+                if (verifyResult.Valid)
+                {
+                    // log any warnings
+                    var warnings = verifyResult.Results.SelectMany(r => r.GetWarningIssues());
+                    if (warnings.Any())
+                    {
+                        foreach (var warning in warnings)
+                        {
+                            packageExtractionContext.Logger.Log(warning);
+                        }
+                    }
+                }
+                else
                 {
                     throw new SignatureException(verifyResult.Results, package);
```

Comment:
This behavior seems inconsistent with the updated NuGetPackageManager behavior above.  Above, if there 1 or more errors and 1 or more warnings, both the warnings are reported.  Here the warnings are not reported if `verifyResult.Valid` is `false`.

Is this correct?  If yes, why?

Reply:
This is intended because All the calls in `PackageExtractor` do not consume the result of the `VerifyPackageSignatureAsync` method. If the method throws a `SignatureException` then the results are handles [here](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Commands/RestoreCommand/ProjectRestoreCommand.cs#L308), [here](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.PackageManagement.PowerShellCmdlets/Cmdlets/InstallPackageCommand.cs#L134) and [here](https://github.com/NuGet/NuGet.Client/blob/dev-anmishr-repobug/src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs#L2343).

But no one is consuming the warnings is the operation was successful. This fixes another issue that @rido-min brought up that warnings do not show up in VS.

Before - 
![capture](https://user-images.githubusercontent.com/10507120/39718314-c4acb6f2-51ea-11e8-906a-bc078c6193cb.PNG)


After - 
![capture2](https://user-images.githubusercontent.com/10507120/39718265-a1355a9e-51ea-11e8-9925-fb9ec96d33e9.PNG)